mlrun/projects/project.py

```diff
@@ -2818,6 +2830,8 @@ def _init_function_from_dict(f, project, name=None):
         func = import_function(url)
         if image:
             func.spec.image = image
+        if name:
+            func.metadata.name = name
```

Comment:
There is also an option to pass the name like this `import_function(new_name=name)` and it will do the trick.. though it is a matter of taste... so up to you